Dockerfile.QA

```diff
@@ -59,20 +59,19 @@ RUN apt-get update && \
             libboost-dev \
             python3-dev \
             python3-pip \
+            python3-wheel \
+            python3-setuptools \
             rapidjson-dev \
             software-properties-common && \
     rm -rf /var/lib/apt/lists/*
 
-RUN pip3 install --upgrade pip && \
-    pip3 install --upgrade wheel setuptools
-
 RUN apt update -q=2 \
     && apt install -y gpg wget \
     && wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - |  tee /usr/share/keyrings/kitware-archive-keyring.gpg >/dev/null \
     && . /etc/os-release \
     && echo "deb [signed-by=/usr/share/keyrings/kitware-archive-keyring.gpg] https://apt.kitware.com/ubuntu/ $UBUNTU_CODENAME main" | tee /etc/apt/sources.list.d/kitware.list >/dev/null \
     && apt-get update -q=2 \
-    && apt-get install -y --no-install-recommends cmake=3.27.7* cmake-data=3.27.7*
+    && apt-get install -y --no-install-recommends cmake=3.28.* cmake-data=3.28.*
```

Comment:
Version 3.28 has a bug with CUDA Toolkit, it was fixed in latest versions.
https://gitlab.kitware.com/cmake/cmake/-/issues/25484

Reply:
The version installed is 3.28.3, which includes the bugfix for above issue - https://github.com/Kitware/CMake/commit/f69b9b7305406f39fda3eff8d60fd30e194e3370

I could pin it further to include patch version, if you think that is better. I don't mind using a recent release as well, just not sure if it will introduce any new issues. What would you recommend?